envoy/tracing/trace_driver.h

```diff
@@ -14,6 +14,37 @@ namespace Tracing {
 class Span;
 using SpanPtr = std::unique_ptr<Span>;
 
+/**
+ * The peer sevice type.
+ */
+enum class ServiceType {
+  // Service type is unknown.
+  Unknown,
+  // Service is treated as HTTP.
+  Http,
+  // Service is treated as GoogleGrpc.
+  GoogleGrpc,
+  // Service is treated as EnvoyGrpc.
+  EnvoyGrpc
+};
+
+/**
+ * The upstream info used for injectContext.
+ */
+struct UpstreamContext {
+  UpstreamContext(const Upstream::HostDescriptionConstSharedPtr& host = nullptr,
+                  const Upstream::ClusterInfoConstSharedPtr& cluster_info = nullptr,
+                  const Tracing::ServiceType service_type = Tracing::ServiceType::Unknown,
+                  const bool is_side_stream = false)
```

Comment:
`cluster_info` is unnecessary here because the `host->cluster()` will return a valid cluster info. (ps: I see. The async client cannot provide a valid host. Hmmm, anyway, please remove the `cluster_info` first until actually users require it.)

And I think the `UpstreamContext` actually doesn't need any ownership. So, maybe a reference to `Upstream::HostDescription` is enough here. If we want to allow null host, then `OptRef<const Upstream::HostDescription>` (Considering there are lots of tests still use the null host, I think this would may be better) could be used.

And please add some comments to explain these parameters, esp, the `is_side_stream`.